components/slidingPanel/slidingPanel.js

```diff
@@ -27,6 +27,12 @@ export default class SlidingPanel extends Component {
     closeOnOverlayClick: PropTypes.bool,
 
     /**
+     * Hook that will be executed when trying to close a panel if exists.
+     * If it returns false, the panel won't be closed.
+     */
+    beforeClosing: PropTypes.func,
```

Comment:
Although the whole code is ok, I'd say that it'd sound better if this method was named `shouldClose` instead of `beforeClosing`, because that name implies a method called before close, nothing else.

Reply:
@froskie had a huge discussion here in Brest, we agreed it that 1) it's like typical hooks: before / after 2) it not just defines should component be closed or not, but also implement custom logic when trying to close. 

I also was thinking about *onTryingToClose*